### PR TITLE
Delete remnant route entries

### PR DESCRIPTION
**Pull request by DockToFuture**
**What this PR does / why we need it**:
Assures that k8s created route entries for gcp shoots are deleted if they are not properly removed by the cloud provider.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:
-->
```improvement operator
Remnant route entries are now explicitly deleted by the gcp-provider extension controller during the infrastructure deletion.
```

### CODE REVIEW

**Comment by DockToFuture** [comment]
> I don't expect any changes in the `contorller-registration.yaml` files, as well as `^` is not a valid Base64 character.
> 
> Please revert those changes.



> I don't expect any changes in the `contorller-registration.yaml` files, as well as `^` is not a valid Base64 character.
> 
> Please revert those changes.

The generate script somehow adds linebreaks which are ```^M``` in base64. Would be nice to found out what causes them when calling ```make generate``` and of course I will remove them.

File: controllers/provider-gcp/pkg/internal/infrastructure/infrastructure.go [hunk 1]
```diff
@@ -56,6 +73,18 @@ func DeleteFirewalls(ctx context.Context, client gcpclient.Interface, projectID
 	return nil
 }
 
+// DeleteRoutes deletes the route entries with the given names in the given project.
+//
+// If a deletion fails, it immediately returns the error of that deletion.
+func DeleteRoutes(ctx context.Context, client gcpclient.Interface, projectID string, routes []string) error {
```
Comment (rfranzke) [inline controllers/provider-gcp/pkg/internal/infrastructure/infrastructure.go]: Please add tests in `infrastructure_test.go`.

File: controllers/provider-gcp/pkg/internal/infrastructure/infrastructure.go [hunk 2]
```diff
@@ -68,6 +97,18 @@ func CleanupKubernetesFirewalls(ctx context.Context, client gcpclient.Interface,
 	return DeleteFirewalls(ctx, client, projectID, firewallNames)
 }
 
+// CleanupKubernetesRoutes lists all Kubernetes route rules and then deletes them one after another.
+//
+// If a deletion fails, this method returns immediately with the encountered error.
+func CleanupKubernetesRoutes(ctx context.Context, client gcpclient.Interface, projectID, network, namespace string) error {
```
Comment (rfranzke) [inline controllers/provider-gcp/pkg/internal/infrastructure/infrastructure.go]: Please add tests in `infrastructure_test.go`.

**Comment by rfranzke** [comment]
> The generate script somehow adds linebreaks which are ^M in base64. Would be nice to found out what causes them when calling make generate and of course I will remove them.

@timuthy suspects that you don't have gnutar. Is that true?

**Comment by DockToFuture** [comment]
> > The generate script somehow adds linebreaks which are ^M in base64. Would be nice to found out what causes them when calling make generate and of course I will remove them.
> 
> @timuthy suspects that you don't have gnutar. Is that true?

For ```make generate``` generate I've used gnutar and locally the files are correctly shown, the newline seperator ```ˆM``` is added when pushing to git.

**Pull request merged**